File: java_db_service/src/main/java/ru/mipt/osint_db_service/repository/webParsing/WebFoundAddressRepository.java
package ru.mipt.osint_db_service.repository.webParsing;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import ru.mipt.osint_db_service.model.webParsing.WebFoundAddress;

@Repository
public interface WebFoundAddressRepository extends JpaRepository<WebFoundAddress, Long> {
    @Query("select wfa from WebFoundAddress wfa order by wfa.foundTime desc")
    Page<WebFoundAddress> getAllOrderByFoundTimeDesc(Pageable pageable);
}

File: Osint_web_service/src/main/kotlin/ru/bitok/osint/osint_web_service/repository/WebFoundAddressRepository.kt
package ru.bitok.osint.osint_web_service.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import ru.bitok.osint.osint_web_service.entity.WebFoundAddress

@Repository
interface WebFoundAddressRepository : JpaRepository<WebFoundAddress, String>{
    fun findByAddress(address: String): WebFoundAddress?

    @Query("""
        select a from WebFoundAddress a
        where a.patternName in (:patternNames) and a.validAddress is null
    """)
    fun findAllNotValidatedInPatternList(patternNames:List<String>): List<WebFoundAddress>
}